### PR TITLE
Box2DGame refactor

### PR DESCRIPTION
**Pull request by spydon**
# Description

Refactor the Box2DGame to make it more straight-forward to use.
Most of this code ended up here instead:
https://github.com/flame-engine/flame_box2d/pull/1

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `v1.0.0` (not `master`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # CHANGELOG
 
 ## [next]
+ - Fix Box2DComponent render priority
```
Comment (luanpotter) [inline CHANGELOG.md:4]: For this particular case should we have a ## 1.0 on top of `next`?

File: lib/box2d/box2d_game.dart [hunk 2]
```diff
@@ -1,47 +1,60 @@
-import 'package:flame/box2d/box2d_component.dart';
-import 'package:flame/box2d/contact_callbacks.dart';
-import 'package:flame/components/component.dart';
+import 'dart:ui';
+
+import 'package:box2d_flame/box2d.dart' hide Timer;
 import 'package:flame/game/base_game.dart';
+import 'package:flame/box2d/viewport.dart';
```
Comment (luanpotter) [inline lib/box2d/box2d_game.dart:5]: existing code, but let's use relative imports for local dependencies.

File: CHANGELOG.md (hunk 1 above)
Comment (spydon) [reply CHANGELOG.md:4]: If we merge into the 1.0 branch and rebase that one on master when we release 0.23.0 I think we should be fine?

File: lib/box2d/box2d_game.dart (hunk 2 above)
Comment (spydon) [reply lib/box2d/box2d_game.dart]: Fixed!

File: lib/box2d/box2d_game.dart [hunk 3]
```diff
@@ -1,47 +1,76 @@
...
   @override
-  void update(double t) {
-    super.update(t);
-    box.components
-        .where((c) => c.destroy())
-        .toList()
-        .forEach((c) => box.remove(c));
-    box.addAll(_addLater);
-    _addLater.clear();
+  void lifecycleStateChange(AppLifecycleState state) {
```
Comment (spydon) [inline lib/box2d/box2d_game.dart:52]: This solves #307 and #296

File: CHANGELOG.md (hunk 1 above)
Comment (spydon) [reply CHANGELOG.md:7]: Fixed

**Pull request merged**